core/src/main/scala/org/apache/spark/Accumulable.scala

```diff
@@ -139,22 +148,28 @@ class Accumulable[R, T] private[spark] (
   }
 
   /**
-   * Set the accumulator's value; only allowed on master
+   * Set the accumulator's value. For internal use only.
    */
-  def setValue(newValue: R) {
-    this.value = newValue
-  }
+  def setValue(newValue: R): Unit = { value_ = newValue }
+
+  /**
+   * Set the accumulator's value. For internal use only.
+   */
+  private[spark] def setValueAny(newValue: Any): Unit = { setValue(newValue.asInstanceOf[R]) }
 
   // Called by Java when deserializing an object
   private def readObject(in: ObjectInputStream): Unit = Utils.tryOrIOException {
     in.defaultReadObject()
-    value_ = zero
-    deserialized = true
+    // TODO: As of SPARK-12895 we send accumulators both ways between executors and the driver.
+    // If we set the value to zero here we would zero out all accumulator updates on the driver,
+    // which is not what we want. Let's comment this out for now until SPARK-12896, which allows
+    // us to avoid sending accumulators from the executors to the driver.
```

Comment:
Shouldn't this be "avoid sending TaskMetrics updates from the executor to the driver"?